components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -1154,7 +1185,17 @@ fn do_upload_package_finish(req: &HttpRequest,
             Ok(pkg) => pkg.visibility,
             Err(_) => {
                 match Origin::get(&ident.origin, &*conn) {
-                    Ok(o) => o.default_package_visibility,
+                    Ok(o) => {
+                        match OriginPackageSettings::create(&NewOriginPackageSettings {
```

Comment:
Is this always a create or do we need to account for update as well?

Reply:
So right now it only calls create on upload and if the entry already exists we "do nothing", basically we only ever use update through a different api which i think is probably the right call.

Reply:
Might be good to add a test for the cases - first upload (create) and second upload (create that does nothing)

Reply:
I believe those tests already exist but I'll see if there's any expansion we can do. We test that behavior in the settings module already I believe.